tests/TestHelpers/InbucketHelper.php

```diff
@@ -0,0 +1,209 @@
+	 * @param int|null $waitTimeSec Time to wait for the email if the email has been delivered
+	 *
+	 * @return string
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public static function getBodyOfLastEmail(
+		string $emailAddress,
+		string $xRequestId,
+		?int $emailNumber = 1,
+		?int $waitTimeSec = EMAIL_WAIT_TIMEOUT_SEC
+	) {
+		$currentTime = \time();
+		$endTime = $currentTime + $waitTimeSec;
+		$mailBox = self::getMailBoxFromEmail($emailAddress);
+		while ($currentTime <= $endTime) {
+			$mailboxResponse = self::getMailboxInformation($mailBox);
+			if (!empty($mailboxResponse) && \sizeof($mailboxResponse) >= $emailNumber) {
+				$mailboxId = $mailboxResponse[\sizeof($mailboxResponse) - $emailNumber]->id;
+				$response = self::getContentOfAnEmailByID($mailBox, $mailboxId, $xRequestId);
+				$body = \str_replace(
+					"\r\n",
+					"\n",
+					\quoted_printable_decode($response->body->text . "\n" . $response->body->html)
+				);
+				return $body;
+			}
+			\usleep(STANDARD_SLEEP_TIME_MICROSEC * 50);
+			$currentTime = \time();
+		}
+		throw new Exception("Could not find the email to the address: " . $emailAddress);
+	}
+
+	/**
+	 * returns body content of a specific email (mailBox) with email ID (mailbox Id)
+	 *
+	 * @param string $mailBox
+	 * @param string $mailboxId
+	 * @param string|null $xRequestId
+	 *
+	 * @return mixed JSON encoded contents
+	 * @throws GuzzleException
+	 */
+	public static function getContentOfAnEmailById(string $mailBox, string $mailboxId, ?string $xRequestId = null) {
+		$response = HttpRequestHelper::get(
+			self::getLocalEmailUrl() . "/api/v1/mailbox/" . $mailBox . "/" . $mailboxId,
+			$xRequestId,
+			null,
+			null,
+			['Content-Type' => 'application/json']
+		);
+
+		return json_decode($response->getBody()->getContents());
+	}
+
+	/**
+	 *
+	 * @param string $emailAddress
+	 * @param string|null $xRequestId
+	 * @param int|null $waitTimeSec Time to wait for the email if the email has been delivered
+	 *
+	 * @return boolean
+	 * @throws Exception
+	 */
+	public static function isEmailReceived(
+		string $emailAddress,
+		?string $xRequestId,
+		?int $waitTimeSec = EMAIL_WAIT_TIMEOUT_SEC
+	):bool {
+		try {
+			self::getBodyOfLastEmail(
+				$emailAddress,
+				$xRequestId,
+				$waitTimeSec
+			);
+		} catch (Exception $err) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * returns the email address of email sender
+	 *
+	 * @param string $emailAddress email address of the receiver
+	 * @param string|null $xRequestId
+	 * @param int|null $emailNumber For email number, 1 means the latest one
+	 * @param int|null $waitTimeSec Time to wait for the email
+	 *
+	 * @return mixed
+	 * @throws GuzzleException
+	 * @throws Exception
+	 */
+	public static function getEmailAddressOfSender(
+		string $emailAddress,
+		string $xRequestId,
+		?int $emailNumber = 1,
+		?int $waitTimeSec = EMAIL_WAIT_TIMEOUT_SEC
+	) {
+		$currentTime = \time();
```

Comment:
as per the function description.
or will it be other than string?
```suggestion
	 * @return string
	 * @throws GuzzleException
	 * @throws Exception
	 */
	public static function getEmailAddressOfSender(
		string $emailAddress,
		string $xRequestId,
		?int $emailNumber = 1,
		?int $waitTimeSec = EMAIL_WAIT_TIMEOUT_SEC
	): string {
		$currentTime = \time();
```

Reply:
yes it will be a string.